### ISSUE TITLE
Failure due to optimistic locking on rook-ceph-osd-orchestration-status with 3 OSDs

### ISSUE THREAD

**Issue by GJKrupa** [rook/rook#1707]
Is this a bug report or feature request?

* Bug Report

**Bug Report**

What happened:
I'm trying to run rook on a small self-hosted cluster.  3 OSDs are configured to use /dev/sdb (a 100GB volume attached to each of 3 worker nodes).

Invariably, one of the OSDs will fail with the following message and then never recover, showing as down when running ceph osd status or ceph osd tree:

`failed to update OSD orchestration status for node k8s-node-3, status {Status:completed Message:}.  Operation cannot be fulfilled on configmaps "rook-ceph-osd-orchestration-status": the object has been modified; please apply your changes to the latest version and try again`

What you expected to happen:
3 OSDs available in the cluster

How to reproduce it (minimal and precise):
Install using the quick start documentation but changing the cluster yaml to the following storage settings:
```
  storage: # cluster level storage configuration and selection
    useAllNodes: true
    useAllDevices: true
    deviceFilter: ^sd[b-z]
    storeConfig:
      storeType: bluestore
```

```
+----+------------+-------+-------+--------+---------+--------+---------+------------+
| id |    host    |  used | avail | wr ops | wr data | rd ops | rd data |   state    |
+----+------------+-------+-------+--------+---------+--------+---------+------------+
| 0  | k8s-node-1 | 1024M | 78.4G |    0   |     0   |    0   |     0   | exists,up  |
| 1  |            |    0  |    0  |    0   |     0   |    0   |     0   | exists,new |
| 2  | k8s-node-2 | 1024M | 78.4G |    0   |     0   |    0   |     0   | exists,up  |
+----+------------+-------+-------+--------+---------+--------+---------+------------+
```

**Environment**:
* OS (e.g. from /etc/os-release): Ubuntu 16.04.4
* Kernel (e.g. `uname -a`): Linux k8s-node-1 4.4.0-31-generic #50-Ubuntu SMP Wed Jul 13 00:07:12 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
* Cloud provider or hardware configuration: ESXi 6.5 i7 16GB RAM 500GB SSD
* Rook version (use `rook version` inside of a Rook Pod): v0.7.0-120.ged7acb9
* Kubernetes version (use `kubectl version`): 1.10.2
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift): kubeadm
* Ceph status (use `ceph health` in the [Rook toolbox](https://Rook.io/docs/Rook/master/toolbox.html)): HEALTH_OK

**Comment by mvasilenko** [comment]
probably hit the same issue here,  
env is different
Cloud provider `OpenStack`, Linux `Ubuntu 16.04.4`, Kernel `4.10.0-28-generic`, Ceph `ceph-12.2.5`, Kubernetes `v1.9.5`, Kubernetes cluster type: `kubespray`

rook-cluster.yaml
```
  storage: # cluster level storage configuration and selection
    useAllNodes: true
    useAllDevices: true
    deviceFilter:
    metadataDevice:
    location:
    storeConfig:
      # The default and recommended storeType is dynamically set to bluestore for devices and filestore for directories.
      # Set the storeType explicitly only if it is required not to use the default.
      # storeType: bluestore
      databaseSizeMB: 1024 # this value can be removed for environments with normal sized disks (100 GB or larger)
      journalSizeMB: 1024  # this value can be removed for environments with normal sized disks (20 GB or larger)
```

```
# ceph status
  cluster:
    id:     3db4b425-706d-49d1-82af-54faf012dc25
    health: HEALTH_OK

  services:
    mon: 3 daemons, quorum rook-ceph-mon1,rook-ceph-mon0,rook-ceph-mon2
    mgr: rook-ceph-mgr0(active)
    osd: 3 osds: 2 up, 2 in

  data:
    pools:   1 pools, 100 pgs
    objects: 0 objects, 0 bytes
    usage:   2052 MB used, 100345 MB / 102397 MB avail
    pgs:     100 active+clean

root@rook-tools:/# ceph osd status
+----+---------------------+-------+-------+--------+---------+--------+---------+------------+
| id |         host        |  used | avail | wr ops | wr data | rd ops | rd data |   state    |
+----+---------------------+-------+-------+--------+---------+--------+---------+------------+
| 0  | rook-ceph-osd-g4h7s | 1026M | 48.9G |    0   |     0   |    0   |     0   | exists,up  |
| 1  |                     |    0  |    0  |    0   |     0   |    0   |     0   | exists,new |
| 2  | rook-ceph-osd-4p9w9 | 1026M | 48.9G |    0   |     0   |    0   |     0   | exists,up  |
+----+---------------------+-------+-------+--------+---------+--------+---------+------------+
```

Expected result: 3 OSD up, actual result: one random OSD is down. If I wipe it's disk, I can connect it later as 4th OSD.

The docker container of failed OSD is goes down during provisioning with the error like in first post. Full log - http://dpaste.com/1T9Y9TW Then, it's get restarted, but unable to join the cluster, because filesystem/block device already marked as busy.

**Comment by grebois** [comment]
Same here:

```
[root@rook-ceph-tools /]# ceph status 
  cluster:
    id:     fc02731b-7203-4db1-b87a-995d869e6c4c
    health: HEALTH_WARN
            Degraded data redundancy: 441/1323 objects degraded (33.333%), 100 pgs degraded, 100 pgs undersized
 
  services:
    mon: 3 daemons, quorum rook-ceph-mon1,rook-ceph-mon2,rook-ceph-mon0
    mgr: a(active)
    osd: 4 osds: 2 up, 1 in
 
  data:
    pools:   1 pools, 100 pgs
    objects: 441 objects, 1159 MB
    usage:   4147 MB used, 1780 GB / 1784 GB avail
    pgs:     441/1323 objects degraded (33.333%)
             100 active+undersized+degraded
 
  io:
    client:   108 kB/s wr, 0 op/s rd, 5 op/s wr
 
[root@rook-ceph-tools /]# ceph osd status
+----+--------------------------------------+-------+-------+--------+---------+--------+---------+------------+
| id |                 host                 |  used | avail | wr ops | wr data | rd ops | rd data |   state    |
+----+--------------------------------------+-------+-------+--------+---------+--------+---------+------------+
| 0  | rook-ceph-osd-1 | 2083M |  890G |    1   |  12.8k  |    0   |     0   | exists,up  |
| 1  | rook-ceph-osd-2 | 2083M |  890G |    1   |   224k  |    0   |     0   | exists,up  |
| 2  |                                      |    0  |    0  |    0   |     0   |    0   |     0   | exists,new |
+----+--------------------------------------+-------+-------+--------+---------+--------+---------+------------+

```

**Comment by travisn** [comment]
This was fixed with #1698, which merged 9 days ago. This status configmap with the contention was changed to be a per-node status so there won't be a conflict anymore
